python/tests/test_relatedness_vector.py

```diff
@@ -0,0 +1,486 @@
+                self.remove_edge(p, c)
+                k += 1
+            while j < M and edges_left[in_order[j]] == left:
+                p = edges_parent[in_order[j]]
+                c = edges_child[in_order[j]]
+                self.insert_edge(p, c)
+                assert self.parent[p] == tskit.NULL or self.x[p] == self.position
+                j += 1
+            right = sequence_length
+            if j < M:
+                right = min(right, edges_left[in_order[j]])
+            if k < M:
+                right = min(right, edges_right[out_order[k]])
+            left = right
+            self.position = left
+
+        # clear remaining things down to virtual samples
+        for j, u in enumerate(self.samples):
+            v = self.virtual_root + 1 + j
+            self.remove_edge(u, v)
+
+        if self.verbosity > 1:
+            self.print_state()
+
+        out = np.zeros(self.sample_weights.shape)
+        for out_i in range(len(self.samples)):
+            i = out_i + self.virtual_root + 1
+            out[out_i] = self.v[i]
+        if self.centre:
+            out -= np.mean(out, axis=0)
+        return out
+
+
+def relatedness_vector(ts, sample_weights, **kwargs):
+    rv = RelatednessVector(
+        sample_weights,
+        ts.num_nodes,
+        samples=ts.samples(),
+        nodes_time=ts.nodes_time,
+        edges_left=ts.edges_left,
+        edges_right=ts.edges_right,
+        edges_parent=ts.edges_parent,
+        edges_child=ts.edges_child,
+        edge_insertion_order=ts.indexes_edge_insertion_order,
+        edge_removal_order=ts.indexes_edge_removal_order,
+        sequence_length=ts.sequence_length,
+        **kwargs,
+    )
+    return rv.run()
+
+
+def relatedness_matrix(ts, centre):
+    Sigma = ts.genetic_relatedness(
+        sample_sets=[[i] for i in ts.samples()],
+        indexes=[(i, j) for i in range(ts.num_samples) for j in range(ts.num_samples)],
+        mode="branch",
+        span_normalise=False,
+        proportion=False,
+        centre=centre,
+    ).reshape((ts.num_samples, ts.num_samples))
+    return Sigma
+
+
+def verify_relatedness_vector(
+    ts, w, *, internal_checks=False, verbosity=0, centre=True
+):
+    R1 = relatedness_vector(
+        ts,
+        sample_weights=w,
+        internal_checks=internal_checks,
+        verbosity=verbosity,
+        centre=centre,
+    )
+    Sigma = relatedness_matrix(ts, centre=centre)
+    R2 = Sigma.dot(w)
+    R3 = ts.genetic_relatedness_vector(w, mode="branch", centre=centre)
+    if verbosity > 0:
+        print(ts.draw_text())
+        print("weights:", w)
+        print("here:", R1)
+        print("with ts:", R2)
+        print("with lib:", R3)
+        print("Sigma:", Sigma)
+    assert R1.shape == (ts.num_samples, w.shape[1])
+    np.testing.assert_allclose(R1, R2, atol=1e-14)
+    # FIXME! Which shape is correct?
+    print(R1.shape, R3.shape)
+    np.testing.assert_allclose(R1.reshape(R3.shape), R3, atol=1e-14)
```

Comment:
```suggestion
    np.testing.assert_allclose(R1, R3, atol=1e-14)
```
Whoops - this snuck in there somehow.